### PR TITLE
Automate Mark II Image Setup WIP

### PR DESCRIPTION
**Pull request by davidwagnerkc**
boot -> wifi -> `./setup.sh` && `source /bin/mycroft-wipe` -> image! 😄 

Well not quite there yet, but that's the idea.

### CODE REVIEW

**Comment by davidwagnerkc** [comment]
@forslund Still WIP, but if you get a chance take a look at `setup.sh` and `mycroft-wipe` let me know if anything seems off. I'll be running through the image creation process many times Wed/Thurs to iron out any final details.

`image_recipe_script.sh` is just a non-tested non-functional draft of converting the `md` file to a script and removing Mark I components that are not necessary. I'll revisit it in a later PR, but wanted to get it committed.

File: home/pi/setup.sh [hunk 1]
```diff
@@ -49,39 +57,53 @@ touch ~/.hushlogin
...
+# Streaming STT
+pip install google-cloud-speech
+# Insert stt key, remove placeholder comment, format and write to file.
+sed '/# Google Service Key/r /boot/stt.json' /etc/mycroft/mycroft.conf \
+    | sed 's|# Google Service.*||' \
+    | python -m json.tool \
+    | sudo tee /etc/mycroft/mycroft.conf
+
+# skills
 ~/mycroft-core/bin/mycroft-msm default
```
Comment (forslund) [inline home/pi/setup.sh]: When msm 0.7.9 hits this should be changed to ~/mycroft-core/bin/mycroft-msm -p mycroft_mark_2pi default to pick the defaults from the correct list.

File: home/pi/setup.sh [hunk 2]
```diff
@@ -22,14 +22,22 @@
...
 
 # Correct permissions from Mark 1 (which used the 'mycroft' user to run)
 sudo chown -R pi:pi /var/log/mycroft
-sudo chmod 666 /var/log/mycroft/*
+rm /var/log/mycroft/*
+sudo chown -R pi:pi /opt/mycroft
+rm -rf /tmp/*
+
+# Locale fix
+sudo sed -i.bak 's|AcceptEnv LANG LC_\*||' /etc/ssh/sshd_config
```
[the remark this answers is not in the document]
Comment (davidwagnerkc) [reply home/pi/setup.sh]: For now I would like to leave them. As we debug things it is nice to be able to do `diff config.txt config.txt.bak` to check things out. We can remove all the backups later.

File: image_recipe_script.sh [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+## Setup packagekit
+apt-get install packagekit
+
+cat <<"EOT" >> /etc/polkit-1/localauthority/50-local.d/allow_mycroft_to_install_package.pkla
+ Identity=unix-user:mycroft
+
+ Action=org.freedesktop.packagekit.package-eula-accept;org.freedesktop.packagekit.package-install
+ ResultAny=yes
+EOT
```
[the remark this answers is not in the document]
Comment (davidwagnerkc) [reply image_recipe_script.sh]: Okay, I'll make a note to look into that. Going to address this script in another PR.

File: image_recipe_script.sh [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+## Allow mycroft user to install with pip
+echo "mycroft ALL=(ALL) NOPASSWD: /usr/local/bin/pip install *" >> /etc/sudoers.d/011_mycroft-nopasswd
+chmod -w /etc/sudoers.d/011_mycroft-nopasswd
```
[the remark this answers is not in the document]
Comment (davidwagnerkc) [reply image_recipe_script.sh]: I'll make a note to check.

File: home/pi/setup.sh (hunk 1 above)
Comment (davidwagnerkc) [reply home/pi/setup.sh]: Sounds good. I'll test that when it is ready

File: home/pi/setup.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (forslund) [reply home/pi/setup.sh]: Understood. Just checking it was considered.

**Comment by davidwagnerkc** [comment]
Still work to do here, but I'm going to merge so that some of the files I have committed can be available. Will start another branch for any final touches.

**Pull request merged**